src/Tribe/Views/V2/Views/By_Day_View.php

```diff
@@ -275,40 +274,45 @@ public function get_grid_days( $date = null, $force = false ) {
 				$day_results   = $repository->by_date( $day_string )->get_set();
 				$day_event_ids = [];
 
-				$event_ids = [];
 				if ( $day_results->count() ) {
 					// Sort events by honoring order and direction.
 					$day_results->order_by( $order_by, $order );
-					$event_ids = array_map( 'absint', $day_results->pluck( 'ID' ) );
 				}
 
 				$this->grid_days_cache[ $day_string ]       = array_values( $day_event_ids );
 				$this->grid_days_found_cache[ $day_string ] = $day_results->count();
 			} else {
-				$start = tribe_beginning_of_day( $day->format( Dates::DBDATETIMEFORMAT ) );
-				$end   = tribe_end_of_day( $day->format( Dates::DBDATETIMEFORMAT ) );
-
-				// Events overlap a day if Event start date <= Day End AND Event end date >= Day Start.
+				$multiday_start = tribe_beginning_of_day( $day->format( Dates::DBDATETIMEFORMAT ) );
+				$multiday_end   = tribe_end_of_day( $day->format( Dates::DBDATETIMEFORMAT ) );
+				$start =  $day->format( 'Y-m-d 00:00:00' ) ;
+				$end   =  $day->format( 'Y-m-d 23:59:59' ) ;
+// @todo Not confident on below changes... site timezone vs event timezone vs timezones assumed on the month view (should be WP or site tz?) and multiday cut off?
+				// Events overlap a day if Event start date <= Day End AND Event end date > Day Start.
 				$results_in_day = array_filter(
 					$day_results,
-					static function ( $event ) use ( $start, $end, $use_site_timezone, $site_timezone, $utc ) {
+					static function ( $event ) use ( $multiday_start, $multiday_end, $start, $end, $use_site_timezone, $site_timezone, $utc ) {
```

Comment:
@bordoni @Camwyn 

For this callback, I was hoping someone could add clarity on it's logic. 

This seems off to me - here we don't use timezones (A):
```
					static function ( $event ) use ( $multiday_start, $multiday_end, $start, $end, $use_site_timezone, $site_timezone, $utc ) {
```
but here we do (B):
```
					static function ( $event ) use ( $multiday_start, $multiday_end, $start, $end, $use_site_timezone, $site_timezone, $utc ) {
```
I don't understand why we do in A but not in B? Seems like a bug...?